### DISCUSSION TITLE
one or more devices is currently unavailable, but ONLINE ????

### DISCUSSION THREAD

**Discussion by korishan** [openzfs/zfs#14215] [Q&A]
I am having a horrible time trying to get this pool back connected.  First a backstory

16TB single drive was created in TrueNAS that was on my Proxmox box installed on a Poweredge R710. The drive was hardware passthrough to the VM.
I migrated from the R710 to an HPE DL380 server. I have a fresh install of pve on that box. I moved the 16TB drive to the HPE. 
Initially I had issues trying to get a new pool showing up in pve (was trying to configure 3x 8TB drives in raidz; eventually it all worked). Later on I was able to successfully import the 16TB drive into the pve pools and I was able to pull the backed up VMs from the old R710 system to the new HPE one. Afterwards I started up TrueNAS and had passed through the drive as well. TrueNAS complained about the drive (not sure of the exact message, but it was like it was partially connected, but not fully as it showed up as available, but couldn't access it to read/write). Then I remembered I hadn't released it from pve host. So I shut down TrueNAS, and went back to pve and issued the export command to drop it off of pve. 
I then went back into TrueNAS and tried to re-import the drive, but it kept failing. 
Because I was worried about some kind of configuration issue because the drive assignments changed, I tried to re-import it back into pve so I could just copy all the data off of it onto the 21TB pool (the 3x 8TB). But I was unable to import it.

I kept getting the following:
```
root@pve-hp1:/etc/zfs# zpool import nas1
cannot import 'nas1': one or more devices is currently unavailable
```

However, if I do zpool import:

```
root@pve-hp1:/etc/zfs# zpool import
   pool: nas1
     id: 14794397138687420911
  state: ONLINE
status: Some supported features are not enabled on the pool.
        (Note that they may be intentionally disabled if the
        compatibility property is set.)
 action: The pool can be imported using its name or numeric identifier, though
        some features will not be available without an explicit zpool upgrade.
 config:

        nas1        ONLINE
          sdl2      ONLINE
```

I can see the pool is there. I have tried using the -f flag, and all sorts of other methods all over the internet forums, but I just can not get this pool to be recognized to connect.

Is there any way to get this thing back online and accessible??  I really would hate to loose the data on the pool.

**Comment by Nachowarrior** [comment]
sudo zpool import -o readonly=on nas1

this worked for me to get a read only version. still looking for a solution to make it writable.

**Reply by oliverhu** [reply]
@Nachowarrior did you find a way to make it writable?

**Reply by Nachowarrior** [reply]
nope. couldn't do ANYTHING with the drive. I don't know if this was the cause of it being in a zpool or what. two identical drives, very low usage. low hours. who knows. i got what i could, now i have one drive because I couldn't find any real help on manipulating zpools in regards to a proper recovery.  I tried at least 15 or 20 different drive recovery tools and multiple methods for each.  odd to me, as usually if you can read a drive, you can do a lot with it.  i'll tell you what though, i'm not buying a samsung ssd ever again as through the process I learned a bit about their shady practices regarding their proprietary drive manipulation software/firmware, which i think in the end is what borked my ability to do more with it. now as far as i am concerned it is junk and i will try to rma it at some point.  sort of unsettling as I have more years than I care to count regarding data recovery and recovering drives on the brink of obliteration. I only posted this as it was the ONLY way to get zfs to show ANYTHING at all. not so sure I like working with zfs from a data integrity standpoint after all this. the underlying ethos of zfs seems to be "throw more hardware at it".
